lib/Graph/Graph.cpp

```diff
@@ -455,29 +456,48 @@ ConvolutionNode *Function::createConv(llvm::StringRef name, NodeValue input,
 }
 
 PoolMaxNode *Function::createPoolMax(llvm::StringRef name, NodeValue input,
-                                     size_t kernel, size_t stride, size_t pad) {
+                                     size_t kernel, size_t stride,
+                                     llvm::ArrayRef<size_t> pads) {
   ShapeNHWC idim = ShapeNHWC(input.dims());
   assert(idim.w >= kernel && idim.h >= kernel &&
          "buffer too small for selected stride");
 
-  auto outSz = calculatePoolOutputDims(idim.h, idim.w, kernel, stride, pad);
+  auto outSz =
+      calculateConvPoolOutputDims(idim.h, idim.w, kernel, stride, pads);
   auto OT = getParent()->uniqueTypeWithNewShape(
       input.getType(), {idim.n, outSz.first, outSz.second, idim.c});
 
-  return addNode(new PoolMaxNode(name, OT, input, kernel, stride, pad));
+  return addNode(new PoolMaxNode(name, OT, input, kernel, stride, pads));
 }
 
-PoolAvgNode *Function::createPoolAvg(llvm::StringRef name, NodeValue input,
+PoolMaxNode *Function::createPoolMax(llvm::StringRef name, NodeValue input,
                                      size_t kernel, size_t stride, size_t pad) {
+  llvm::SmallVector<size_t, 4> pads = {pad, pad, pad, pad};
+  return createPoolMax(name, input, kernel, stride, pads);
+}
+
+PoolAvgNode *Function::createPoolAvg(llvm::StringRef name, NodeValue input,
+                                     size_t kernel, size_t stride,
+                                     llvm::ArrayRef<size_t> pads) {
   ShapeNHWC idim = ShapeNHWC(input.dims());
-  assert(idim.w >= kernel && idim.h >= kernel &&
+  PaddingTLBR pdim(pads);
+  (void)pads;
```

Comment:
This `(void)pads;` shouldn't be necessary since `pads` is used in non-debug code... right?  Oh, wait -- I think you want `(void)pdim`, since that's only used in the assertion.